pkg/cloudprovider/nodecontroller/nodecontroller.go

```diff
@@ -253,6 +257,12 @@ func (nc *NodeController) Run(period time.Duration, syncNodeList bool) {
 			glog.Errorf("Error monitoring node status: %v", err)
 		}
 	}, nc.nodeMonitorPeriod)
+
+	go util.Forever(func() {
```

Comment:
I believe this is clearer than asking API server for list of pods. It's probably a matter of taste, but I'd rather depend on local thread than external service.